plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

```diff
@@ -751,7 +751,17 @@ jQuery( function ( $ ) {
 					}
 				);
 				selectedAttributes = newSelectedAttributes;
-				$( 'select.wc-attribute-search' ).data('disabled-items', newSelectedAttributes );
+				$( 'select.wc-attribute-search' ).data(
+					'disabled-items',
+					newSelectedAttributes
+				);
+				var isUsedForVariations = $( 'input#used-for-variation' ).is(
+					':checked'
+				);
+				$( 'select.wc-attribute-search' ).data(
+					'add-attribute-used-for-variations',
```

Comment:
Nitpicky, but could we make this `is-used-for-variations` which defines state for this item instead of an intent?

Reply:
Good idea; I changed it in the commit [aa0f99b](https://github.com/woocommerce/woocommerce/pull/35807/commits/aa0f99b2a88faed0d1580d561eaf2e19c679ab93)